server/src/main/java/org/opensearch/index/translog/TranslogWriter.java

```diff
@@ -189,12 +189,12 @@ public static TranslogWriter create(
             checkpointChannel = channelFactory.open(checkpointFile, StandardOpenOption.WRITE);
             final TranslogHeader header = new TranslogHeader(translogUUID, primaryTerm);
             header.write(channel, !Boolean.TRUE.equals(remoteTranslogEnabled));
-            TranslogCheckedContainer translogCheckedContainer = null;
-            if (Boolean.TRUE.equals(remoteTranslogEnabled)) {
-                ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-                header.write(byteArrayOutputStream);
-                translogCheckedContainer = new TranslogCheckedContainer(byteArrayOutputStream.toByteArray());
-            }
+
+            // Enable translogCheckedContainer for remote as well as local translog.
```

Comment:
Why are we enabling `translogCheckedContainer` for local translogs?

Reply:
As per discussion with @Bukhtawar and @sachinpkale, for ensuring consistent view of the world, we are enabling footer for both local and remote translogs.

Reply:
I am assuming there would be low/minimal overhead of checksum for the local translog? Can we run a perf benchmark?

Reply:
Yes, I am running the Mensor test for the same to understand the impact. But from code workflow POV, it seems to be a low overhead operation.